android/app/src/main/java/app/covidshield/services/metrics/UniqueDailyDebugMetricsHelper.kt

```diff
@@ -0,0 +1,40 @@
+package app.covidshield.services.metrics
+
+import android.content.Context
+import java.util.*
+
+object UniqueDailyDebugMetricsHelper {
+
+    private const val SharedPreferencesFileKey = "covid-shield-unique-daily-debug-metrics"
+
+    fun canPublishMetric(metricIdentifier: String, context: Context): Boolean {
+        val sharedPreferences = context.getSharedPreferences(SharedPreferencesFileKey, Context.MODE_PRIVATE)
+
+        return if (sharedPreferences.contains(metricIdentifier)) {
+            val retrievedUTCDateInMilliseconds = sharedPreferences.getLong(metricIdentifier, 0)
+            !isSameDay(Date(retrievedUTCDateInMilliseconds), Date(System.currentTimeMillis()))
+        } else {
+            true
+        }
+    }
+
+    fun markMetricAsPublished(metricIdentifier: String, context: Context) {
+        val sharedPreferences = context.getSharedPreferences(SharedPreferencesFileKey, Context.MODE_PRIVATE)
+
+        with (sharedPreferences.edit()) {
+            putLong(metricIdentifier, System.currentTimeMillis())
+            apply()
+        }
+    }
+
+    private fun isSameDay(date1: Date, date2: Date): Boolean {
```

Comment:
It would be nice if this function was in a class on it's own, or with some other date functions.

Reply:
@jeberhardt I put it here because I knew you were going to have some kind of a Date class at some point with the Kotlin rewrite but since we do not know if it is going to happen any time soon then I am going to extract it.